clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsRequestManager.java

```diff
@@ -116,15 +138,27 @@ public OffsetsRequestManager(final SubscriptionState subscriptionState,
         this.subscriptionState = subscriptionState;
         this.time = time;
         this.requestTimeoutMs = requestTimeoutMs;
+        this.defaultApiTimeoutMs = defaultApiTimeoutMs;
         this.apiVersions = apiVersions;
         this.networkClientDelegate = networkClientDelegate;
-        this.backgroundEventHandler = backgroundEventHandler;
         this.offsetFetcherUtils = new OffsetFetcherUtils(logContext, metadata, subscriptionState,
                 time, retryBackoffMs, apiVersions);
         // Register the cluster metadata update callback. Note this only relies on the
         // requestsToRetry initialized above, and won't be invoked until all managers are
         // initialized and the network thread started.
         this.metadata.addClusterUpdateListener(this);
+        this.commitRequestManager = commitRequestManager;
+    }
+
+    private static class PendingFetchCommittedRequest {
+        final Set<TopicPartition> requestedPartitions;
+        final CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> result;
+
+        private PendingFetchCommittedRequest(final Set<TopicPartition> requestedPartitions,
+                                              final CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> result) {
```

Comment:
Super-duper nit: parameter alignment:

```suggestion
                                             final CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> result) {
```